### DISCUSSION TITLE
current heading

### DISCUSSION THREAD

**Discussion by akifbora** [rnmapbox/maps#3727] [Q&A] [answered]
Is there any method that I can get the current heading (bearing) value?

**Comment by Gnative** [comment]
It is not documented but it is possible to subscribe to the locationManager which returns an event with a heading. It is used in the UserLocation component 

```js
import { locationManager } from '@rnmapbox/maps'

const onLocationHandler = (event) => {
    if(event?.coords?.heading){
        //
    }
}

locationManager.addListener(onLocationHandler)
locationManager.start()

locationManager.removeListener(onLocationHandler)
// careful when stopping the location manager as this can stop
// the user location update in a map view if it is being used
locationManager.stop()
```

**Reply by akifbora** [reply]
thanks for the reply. but is it possible without location manager? something like map.getBearing() in mapbox-gl-js?

**Reply by Gnative** [reply]
I'm not sure. You'll have to dig through the code to see, else there are some other location packages you could use.

- [@react-native-community/geolocation](https://www.npmjs.com/package/@react-native-community/geolocation)
- [expo-location](https://docs.expo.dev/versions/latest/sdk/location/)

**Reply by akifbora** [reply]
I think I was misunderstood. We can set heading like `cameraRef.setCamera({ heading: 30 })`, but I don't see any method to get the current heading/bearing value. Has the camera class any method to get map's current bearing value without location data?
Please refer:
https://docs.mapbox.com/ios/maps/guides/camera-and-animation/camera/#get-camera-position

Sorry for taking your time.

**Comment by akifbora** [comment, marked as answer]
I got it. there is no method to get current bearing and pitch on mapRef. I implemented and I will send a pull request if needed.
